R/System_Expression.R

```diff
@@ -0,0 +1,56 @@
+# System_Expression.R
+
+#' System level expression values.
+#'
+#' \code{exProf} Returns the log-normalized expression values for gene provided.
+#'
+#' \code{System_Expression} Provides the mean expression level for each experiment
+#' present in the dataset. As well, displays which conditions are up and down
+#' regulted significantly.
+#'
+#' For further information, see details in the \href{../doc/BCB420.2019.ESA-Vignette_McNeil.html}{ System Expression vignette} (or load the vignette with \code{vignette("BCB420.2019.ESA-Vignette_McNeil", package = "BCB420.2019.ESA")}).
+#'
+#' @param sysname (character) 5-letter code of system being investigated
+#' @param myQNXP (matrix) Matrix of expressions. Columns represent experiments and
+#' rows individual genes
+#'
+#' @return (list) A list containing 3 elements. The first the mean log normalized
+#' gene expression by condition. Second and third the conditions where the system
+#' is significantly up and down regulated respectively.
+#'
+#' @author \href{https://orcid.org/0000-0002-2821-6204}{Matthew McNei;} (aut)
+#'
+#' @export
+
+System_Expression<- function(sysname, myQNXP){
+  sysDB <- fetchData("SysDB")
+  # Get a list of genes present in the given system
+  components <- SyDBgetSysSymbols(sysDB, sysname)[[1]]
+
+  # Normalized expression values for each gene in the system
+  expressions <- sapply(components, exProf, myQNXP)
+
+  # Find the means and variance of the expression changes caused by each condition
+  meanExpression <- rowMeans(expressions, na.rm = T)
+  varExpression <- apply(expressions, 1, var, na.rm = T)
+  num_observations <- apply(expressions, 1, function(x) sum(!is.na(x)))
+
+  # calculate the one-sided p-values for each experiment
+  p.values <- pnorm(meanExpression, sd = sqrt(varExpression/num_observations))
```

Comment:
Coding style - variable names should not contain "."

Please see
http://steipe.biochemistry.utoronto.ca/abc/index.php/RPR-Coding_style

Reply:
Fixed